src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx

```diff
@@ -133,6 +134,12 @@ const CreateTokenTransfer = ({
   const isDisabled = isSafeTokenSelected && isSafeTokenPaused
   const isAddressValid = !!recipient && !errors[TokenTransferFields.recipient]
 
+  useEffect(() => {
+    if (!spendingLimit) {
+      setNonceNeeded(true)
+    }
```

Comment:
Will it flip back if they switch to a spending limit?

Reply:
Yes the `SpendingLimitRow` `onChange` handler also sets setNonceNeeded.